File: casper/src/test/scala/coop/rchain/casper/genesis/contracts/StackSpec.scala
package coop.rchain.casper.genesis.contracts
import coop.rchain.casper.helper.RhoSpec
import coop.rchain.models.NormalizerEnv
import coop.rchain.rholang.build.CompiledRholangSource

class StackSpec
    extends RhoSpec(
      CompiledRholangSource("StackTest.rho", NormalizerEnv.Empty),
      Seq.empty,
      GENESIS_TEST_TIMEOUT
    )

File: casper/tests/genesis/contracts/stack_spec.rs
// See casper/src/test/scala/coop/rchain/casper/genesis/contracts/StackSpec.scala

use crate::genesis::contracts::GENESIS_TEST_TIMEOUT;
use crate::helper::rho_spec::RhoSpec;
use rholang::rust::build::compile_rholang_source::CompiledRholangSource;
use std::collections::HashMap;

#[tokio::test]
async fn stack_spec() {
    let test_object =
        crate::util::rholang::test_rho_loader::load_test_rho("StackTest.rho").expect("Failed to load StackTest.rho");

    let compiled = CompiledRholangSource::new(
        test_object,
        HashMap::new(), // NormalizerEnv.Empty
        "StackTest.rho".to_string(),
    )
    .expect("Failed to compile StackTest.rho");

    let spec = RhoSpec::new(compiled, vec![], GENESIS_TEST_TIMEOUT);

    spec.run_tests().await.expect("StackSpec tests failed");
}
